test/aws/ec2.yml

```diff
@@ -0,0 +1,72 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: >-
+  A stack for testing AWSCore on EC2.
+
+Parameters:
+  MyKeyName:
+    Description: The keyname associated with the instance.
+    Type: String
+    Default: ec2_keyfile
+
+Resources:
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: AWSCore-EC2-Test
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+  InstanceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+  InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Roles:
+        - !Ref InstanceRole
+  TestingEC2:
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: t2.micro
+      ImageId: ami-0922553b7b0369273
+      # Instance must have a profile for `ec2_instance_credentials()` to succeed
```

Comment:
What happens if it doesn't? Where would credentials be found then? Would they just not exist?

Reply:
Credentials would not exist for the instance without a profile. Currently, the `ec2_instance_credentials` function will error due to the instance metadata not having an `iam` path if there is no profile.

Reply:
What I mean to ask is, what would happen if you called `aws_config()` in that scenario?

Reply:
Cool, thanks